platform/SingletonPtr.h

```diff
@@ -105,6 +105,16 @@ struct SingletonPtr {
         return get();
     }
 
+    /** Get a reference to the underlying singleton
+     *
+     * @returns
+     *   A reference to the singleton
+     */
+    T &operator*()
```

Comment:
Isn't this taken care of by the `T *operator->()` operator?

Reply:
No - the alternative here would be `lock(*_mutex.get())`, which seemed unnecessarily awkward.

Reply:
Ah you're right, I though operator-> was full magic, but it's only half magic :+1:

Reply:
That does cover the "I want a reference" case, but I wonder if we should also have a conversion operator to `T *` for people who want the pointer. Does the `get()` really have to be explicit? Is there any C++ trap there for the conversion operator?